### PR TITLE
Update Developer Guide

### PR DESCRIPTION
**Pull request by potty10**
This commit 
- Fixes all figures related to `delete-field` and `delete-index`
- Update implementation section for  `delete-field` 
- Propose an enhancement to fix issue #169

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -240,18 +240,18 @@ How the `Logic` component works:
 
 Figure 7 below illustrates the interactions within the `Logic` component for the `execute("delete-index 1")` API call.
 
-![Interactions Inside the Logic Component for the `delete 1` Command](images/DeleteSequenceDiagram.png)
+![Interactions Inside the Logic Component for the `delete 1` Command](images/DeleteIndexSequenceDiagram.png)
```
Comment (eugenetangkj) [inline docs/DeveloperGuide.md:243]: This line should also change according to the command that you are trying to depict, as mentioned in line 241.

File: docs/diagrams/DeleteFieldSequenceDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,90 @@
+@startuml
```
Comment (eugenetangkj) [inline docs/diagrams/DeleteFieldSequenceDiagram.puml:1]: Are the following parts considered as mistakes? (Refer to image for the numbering)

1. The activation bar should be connected to the class name?
2. Should `p` be returned from the constructor?
3. Should `c` be returned from the constructor?
4. Are there missing dotted arrows to denote the return from `model`?
5. Should `cr` be returned from `LogicManager`?


image

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -240,18 +240,18 @@ How the `Logic` component works:
 
 Figure 7 below illustrates the interactions within the `Logic` component for the `execute("delete-index 1")` API call.
```
Comment (eugenetangkj) [inline docs/DeveloperGuide.md:241]: The description is for `delete-index 1` but your sequence diagram is for `delete 1-2`. Shall we fix it to be the same?

File: docs/diagrams/DeleteIndexSequenceDiagram.puml [hunk 4]
```diff
@@ -0,0 +1,81 @@
+@startuml
```
Comment (eugenetangkj) [inline docs/diagrams/DeleteIndexSequenceDiagram.puml:1]: Similar issues raised in my comment for `DeleteFieldSequenceDiagram`.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -1484,17 +1474,30 @@ Assumptions: The sample data provided by InternBuddy is used, where there is a t
...
+
+ For example, using the `find` command, `find t/numbers t/cheese` tries to find entries with either the tag `numbers` or `cheese`. Tag matching is case-insensitive, so it also tries to find `Numbers` or `CheEse`. Similarly, `delete-field t/numbers t/cheese` tries to delete entries with either the tag `numbers` or `cheese` or `Numbers` or `CheEse`
+
+There could be another conflicting meaning of the command. In this case, `find t/numbers t/cheese` tries to find "numbers" or "cheese". but the user could have intended to find the tag `numbers T/cheese`. This could lead to some confusion.
+
+**Proposed Design Tweak**: Make prefixes case-insensitive. For example, `find t/numbers t/cheese` should have the same result as `find T/numbers T/cheese`. This would remove the confusion as substrings such as `T/` or `c/` cannot be entered in any of the fields. 
 
-**Proposed Design Tweak**: Explain the design tweak here
+Alternatively, we propose to implement a warning if strings such as ` T/` is detected in the user input. It is unlikely user intended to add `numbers T/cheese` as a tag and if the user added such a tag by mistake, the remedy is for the user to edit it.
```
Comment (eugenetangkj) [inline docs/DeveloperGuide.md:1500]: Would it be difficult to provide a bit more concrete implementation on how to make prefixes case-insensitive and/or how to implement the warning?

For example, which part of the code needs adjustment and/or what kind of warning message would be displayed to the user.

If it's too troublesome then it's okay.

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -1484,17 +1474,30 @@ Assumptions: The sample data provided by InternBuddy is used, where there is a t
...
+   and tag `Golang T/c++`.
+  Therefore, it is possible to add substrings such as `T/`, `C/` or `R/` to any of the fields, even though the user could have intended to enter `t/`, `c/` or `r/`. 
+
+Moreoever, commmands such as `find` and `delete-field` do not consider prefixes such as `c/`, although other commands such as `edit` and `add` do  use the prefix `c/`. In the case of `find` and `delete-field`, it is possible to add the substring `c/` to any of the fields, even though the user could have intended to enter `c/` as a command prefix.
+
+We originally intended for this design to allow maximum flexibility for the user. If the user had entered something wrongly, it is possible to just use the command `edit`. However, this design could have an unintentional side-effect.
+
+ For example, using the `find` command, `find t/numbers t/cheese` tries to find entries with either the tag `numbers` or `cheese`. Tag matching is case-insensitive, so it also tries to find `Numbers` or `CheEse`. Similarly, `delete-field t/numbers t/cheese` tries to delete entries with either the tag `numbers` or `cheese` or `Numbers` or `CheEse`
```
Comment (eugenetangkj) [inline docs/DeveloperGuide.md:1494]: Do you think it's more appropriate to use tags that are related to our project instead of `cheese`?

**Review by eugenetangkj — COMMENTED** [review]
Great effort! I have left some comments, you can decide whether they are appropriate to be changed or not 👍

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (potty10) [reply docs/DeveloperGuide.md:241]: The sequence diagram shows `delete-index 1 2`

File: docs/diagrams/DeleteFieldSequenceDiagram.puml (hunk 2 above)
Comment (potty10) [reply docs/diagrams/DeleteFieldSequenceDiagram.puml:1]: Thanks for the review. I have a problem with point 1. If I split the constructor into 2 lines (`internshipContainsKeywordPredicate(...)`), there will be a gap. I had to add a new line to the blue box. Is it ok?;

image

Comment (eugenetangkj) [reply docs/diagrams/DeleteFieldSequenceDiagram.puml:1]: Yes, it looks great! 👍

**Comment by eugenetangkj** [comment]
Looks good to merge! Thanks for the effort

**Pull request merged**